pkg/util/timeutil/ptp/ptp_clock_linux_test.go

```diff
@@ -28,3 +31,29 @@ func TestClockNow(t *testing.T) {
 		t.Errorf("clock mismatch: got %v; timeutil says %v", got, want)
 	}
 }
+
+func TestInitClock(t *testing.T) {
+	var file *os.File
+	var err error
+	var devOpenerCallCount int
+	clock := &Clock{
+		devOpener: func(name string) (*os.File, error) {
+			require.Equal(t, "/dev/ptp-test", name)
+			file, err = os.CreateTemp("", "")
+			require.NoError(t, err)
+			require.NotNil(t, file)
+			devOpenerCallCount++
+			return file, err
+		},
+		cGetTime: func(fd fdType, ts *structTimespec) error {
```

Comment:
The first parameter should be of type `C.clockid_t`, and should probably be renamed to something like `clockID`.

It's probably the same int type under the hood, but semantics are important. File descriptors are constants returned by OS. Clock ID is derived from a file descriptor, see the `Fd()<<3|3 == fd` below.